sql/parser/datum.go

```diff
@@ -337,6 +337,12 @@ type DDate struct {
 	time.Time
 }
 
+// MakeDDate constructs a DDate from a time.Time.
+func MakeDDate(t time.Time) DDate {
+	year, month, day := t.Date()
+	return DDate{Time: time.Date(year, month, day, 0, 0, 0, 0, t.Location())}
```

Comment:
Actually, that is not equivalent; `Truncate` always truncates to UTC, while this version is timezone-aware.